### PR TITLE
feat(js-beautifier): using js-beautify for formatting html

### PR DESCRIPTION
**Pull request by JayaKrishnaNamburu**
Discussion regarding #292

### CODE REVIEW

File: packages/teleport-postprocessor-js-beautify-html/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+// @ts-ignore
```
Comment (alexnm) [inline packages/teleport-postprocessor-js-beautify-html/src/index.ts]: let's check if @types/js-beautify exists so we can have typing support for all dependencies

File: packages/teleport-postprocessor-js-beautify-html/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  formatOptions?: PrettierFormatOptions
+}
+
+export const createPostProcessor = (options: PostProcessorFactoryOptions = {}) => {
+  const fileType = options.fileType || FileType.HTML
+
+  const processor: PostProcessor = (codeChunks) => {
+    if (codeChunks[fileType]) {
+      const result = beautifyHTML(codeChunks[fileType], {
+        wrap_line_length: 80,
```
Comment (alexnm) [inline packages/teleport-postprocessor-js-beautify-html/src/index.ts]: can we extract this into constants? same as prettier settings

File: packages/teleport-postprocessor-js-beautify-html/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        end_with_newline: true,
+        preserve_newlines: true,
+      })
+      codeChunks[fileType] = result
+    } else {
+      console.warn('No code chunk of type HTML found, prettier-html did not perform any operation')
```
Comment (alexnm) [inline packages/teleport-postprocessor-js-beautify-html/src/index.ts]: make sure you rename the message here

File: packages/teleport-postprocessor-js-beautify-html/src/index.ts (hunk 2 above)
Comment (JayaKrishnaNamburu) [reply packages/teleport-postprocessor-js-beautify-html/src/index.ts]: yup 👍

File: packages/teleport-postprocessor-js-beautify-html/src/index.ts (hunk 3 above)
Comment (JayaKrishnaNamburu) [reply packages/teleport-postprocessor-js-beautify-html/src/index.ts]: Yeah yeah 😅  this was for testing purpose. I just copied 😅

**Comment by JayaKrishnaNamburu** [comment]
Hey @alexnm i noticed this --> https://github.com/teleporthq/teleport-code-generators/blob/master/packages/teleport-postprocessor-prettier-js/src/index.ts#L21

Here is my opinion, currently we are using prettier for formatting and so we got the flexibility to parse js and jsx content both with babel. But when we are trying to add more formatters, I noticed this behavior.

1. A pure JS Formatter can't format a JSX file (does not break, but formats it very poorly)
2. For example, let us take the behavior of *project-generator-react* the output file is *js* but actually it has `jsx` inside it. Which babel takes care out of the box but other pure JS formatters tend to fail. 
- I feel we need to have something like a content flag, which describes the type of content inside the file. So then we can do something like this --> https://github.com/teleporthq/teleport-code-generators/pull/370/files#diff-3a9f5f76c2ea1b8e9388bb72e21959d6R7

**Comment by alexnm** [comment]
closing this as it doesn't work out with what we need

**Pull request closed without merging**